### ISSUE TITLE
Desktop App

### ISSUE THREAD

**Issue by vikasgola** [Sangwan5688/BlackHole#74]
It would be cool and very helpful if we can have this on desktop(Linux/Windows) as well. Because, app has been made in flutter and flutter support desktop apps, it should be fairly easy to port it to desktop.

**Comment by brnwlshubh** [comment]
https://github.com/Sangwan5688/BlackHole/issues/38#issuecomment-860154817 Refer this

**Comment by Sangwan5688** [comment]
As mentioned by @brnwlshubh I have already answered this request in #38. The app can be extended for Desktop, but there are still a lot of things that need to be done on the Android Side. Only after all those things, I will try expanding the app to other platforms. Also, the app does use some packages which aren't available for other platforms. I will keep this issue open so that If anyone else wants to extend the app for desktop, he/she can do so. PRs regarding the issue are welcomed :)

**Comment by Sangwan5688** [comment]
I'm attaching a alpha version of BlackHole for windows with this comment. As it's the very first alpha version nothing will work except download. Even download will be done without any kind of metadata. So, if you just want to download song and can edit metadata manually then you can try it.

[bh-win.zip](https://github.com/Sangwan5688/BlackHole/files/7109706/bh-win.zip)

**Comment by naqushab** [comment]
Can you provide instructions to build the app on desktop @Sangwan5688 , thanks!! A web version of this would also work.

**Comment by Sangwan5688** [comment]
1. Clone the Repo
2. Make sure flutter and Dart SDK are installed.
3. Run Pub get and all to get all dependencies.
4. Make sure you have `Desktop Development with C++` workload installed (it is required for Desktop Development). It comes with visual studio, so just install vs community or whatever you like and check `Desktop Development with C++` in workloads.
4. Run `flutter build windows` and it will build for windows. Built files will be available in `build/windows/runner/release`
5. So you can finally open that exe file to run the app. It will create a portable version of the app where installation isn't required. If you can to create a single setup file. Just run `flutter pub run msix:create`. It will create a single msix file then.

**Comment by hackingguy** [comment]
```
[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: MissingPluginException(No implementation found for method init on channel com.ryanheise.just_audio.methods)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:175)
<asynchronous suspension>
#1      MethodChannelJustAudio.init (package:just_audio_platform_interface/method_channel_just_audio.dart:13)
<asynchronous suspension>
#2      AudioPlayer._setPlatformActive.setPlatform (package:just_audio/just_audio.dart:1255)
<asynchronous suspension>

[ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: MissingPluginException(No implementation found for method init on channel com.ryanheise.just_audio.methods)
#0      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:175)
<asynchronous suspension>
#1      MethodChannelJustAudio.init (package:just_audio_platform_interface/method_channel_just_audio.dart:13)
<asynchronous suspension>
#2      AudioPlayer._setPlatformActive.setPlatform (package:just_audio/just_audio.dart:1255)
<asynchronous suspension>
```
System Info:
```
Gnome: 41
WM: Wayland
Theme: WhiteSur-gtk-theme
Distro: Manjaro 21.2.0
Kernel: 5.15.7-1
```

**Comment by Sangwan5688** [comment]
It's been some time since BlackHole was compiled for windows. I forgot to add it here as there were some issues in uploading the files on github. Anyway, If anyone want the windows version, he/she can get it from the [telegram channel](https://t.me/blackhole_official/106). The one posted there is a portable version, if you want an installable `msix` file then even that is available, just comment there, I will provide it to you. I will even post the windows, Linux and iOS versions on GitHub with next release.

**Comment by naqushab** [comment]
I have a MacOS machine, if that helps, I can compile the project and provide binaries. I need to know what all things are required to build.

**Comment by Sangwan5688** [comment]
@naqushab most probably there will be some errors the first time you try it. But it should work just fine after fixing those few errors. About how to compile it for macos, you can look over the [flutter website](https://docs.flutter.dev/desktop). Just try following the instruction and if you have and issue, create a new ticket for the same (focused on macOS) as this one is related to windows to be specific
